components/beehiiv/app/beehiiv.app.ts

```diff
@@ -38,27 +38,32 @@ export default defineApp({
     },
   },
   methods: {
-    _getUrl(path) {
-      return `https://api.beehiiv.com/v1${path}`;
+    _getUrl(path, params = {}) {
+      let formattedPath = path;
+      for (const [key, value] of Object.entries(params)) {
+        formattedPath = formattedPath.replace(`{${key}}`, value);
+      }
+      return `https://api.beehiiv.com/v2${formattedPath}`;
     },
     _getHeaders(headers = {}) {
       return {
-        "X-ApiKey": this.$auth.api_key,
+        "Authorization": `Bearer ${this.$auth.api_key}`,
         "Content-Type": "application/json",
         ...headers,
       };
     },
     _getRequestParams(opts: any) {
       return {
         ...opts,
-        url: this._getUrl(opts.path),
+        url: this._getUrl(opts.path, opts.params),
         headers: this._getHeaders(),
       };
     },
     async createSubscriber($ = this, param) {
       const response = await axios($, this._getRequestParams({
         method: "POST",
-        path: "/subscribers",
+        path: "/publications/{publicationId}/subscriptions",
+        params: { publicationId: param.publication_id },
         data: param,
```

Comment:
Is it ok for `publication_id` to be included in the request body as well, since it is being used as a query param?

Reply:
Yes, I think so!